### PR TITLE
JavaScript Exercises & Mandatory Complete

### PR DESCRIPTION
**Pull request by SimonLawrenceGB**
Next up mandatory.

**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Simon Lawrence
- Your City: London
- Your Slack Name: Simon - Trainee

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: exercises/F-strings-methods/exercise.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
 // Start by creating a variable `message`
+var greetingStart = "My name is ";
+var greetingMiddle = " and my name is ";
+var greetingEnd = " characters long";
+var myName = "Simon";
+var nameLength = myName.length;
 
-console.log(message);
+console.log(greetingStart + myName + greetingMiddle + nameLength + greetingEnd);
```
Comment (jstadnik) [inline exercises/F-strings-methods/exercise.js:8]: Could also use string interpolation
```suggestion
console.log(`My name is ${myName} and my name is ${nameLength} characters long`);
```

File: exercises/G-numbers/exercise.js [hunk 2]
```diff
@@ -1 +1,10 @@
 // Start by creating a variables `numberOfStudents` and `numberOfMentors`
+let numStu = "Number of Students: ";
+let numMen = "Number of Mentors: ";
+let numStuMen = "Total Number of Students and Mentors: ";
+let numberOfStudents = 15;
+let numberOfMentors = 8;
+
+console.log(numStu + numberOfStudents);
```
Comment (jstadnik) [inline exercises/G-numbers/exercise.js:8]: What you did is just fine, but also you don't have to declare parts of the string as a variable, especially if you're only using it once. So here you could simply do:
```javascript
console.log("Number of students: " + numberOfStudents);
```

File: exercises/K-functions-parameters/exercise4.js [hunk 3]
```diff
@@ -1,5 +1,9 @@
 // Declare your function first
+function numbers(a, b) {
```
Comment (jstadnik) [inline exercises/K-functions-parameters/exercise4.js:2]: Small nitpick, but "numbers" is not a very descriptive function name, "add" might be better. (You'd be surprised how non-trival amount of coding is coming up with good names...)

File: exercises/L-functions-nested/exercise.js [hunk 4]
```diff
@@ -3,3 +3,25 @@ var mentor2 = "Irina";
 var mentor3 = "Mimi";
 var mentor4 = "Rob";
 var mentor5 = "Yohannes";
+
+function shout(hello) {
+  return hello;
+}
```
Comment (jstadnik) [inline exercises/L-functions-nested/exercise.js:9]: So the exercise asks for one function that converts the mentor name to upper case and the other one that creates a shouty greeting. So if I gave you a function:

```javascript
// Converts name to upper case
function shout(name) {
    return name.toUpperCase();
}
```
And asked you to create a shoutyGreeting such that I could call it like this:
```node
>>> shoutyGreeting("Daniel")
>>> "HELLO DANIEL"
```

How would you write it?

**Review by jstadnik — COMMENTED** [review]
Good start on the exercises, try doing the mandatory ones as well :)

File: exercises/L-functions-nested/exercise.js (hunk 4 above)
Comment (SimonLawrenceGB) [reply exercises/L-functions-nested/exercise.js:9]: Tried to figure this out but couldn't quite get it.

Comment (SimonLawrenceGB) [reply exercises/L-functions-nested/exercise.js]: I did by looking at an example online but still don't fully understand how it works.

File: mandatory/3-function-output.js [hunk 5]
```diff
@@ -1,16 +1,17 @@
...
 // Add comments to explain what this function does. You're meant to use Google!
 function combine2Words(word1, word2) {
   return word1.concat(word2);
-}
+} // This function combines two different words into one. The end result being that "word1" and "word2" now become "word1word2".
 
 function concatenate(firstWord, secondWord, thirdWord) {
   // Write the body of this function to concatenate three words together.
   // Look at the test case below to understand what this function is expected to return.
+  return firstWord.concat(secondWord) + secondWord.concat(thirdWord);
```
Comment (jstadnik) [inline mandatory/3-function-output.js:14]: So what went wrong here is when you evaluate the two expressions, you're actually returning this
```javascript
return "firstWordsecondWord" + "secondWordthirdWord"
```
If that makes sense?

You want something more akin to 
```javascript
return firstWord.concat(secondWord).concat(thirdWord)
```
And additionally some spaces.

File: mandatory/4-tax.js [hunk 6]
```diff
@@ -17,7 +19,10 @@ function calculateSalesTax() {}
   Remember that the prices must include the sales tax (hint: you already wrote a function for this!)
 */
 
-function addTaxAndFormatCurrency() {}
+function addTaxAndFormatCurrency(cost) {
+  let costIncTax = calculateSalesTax(cost);
```
Comment (jstadnik) [inline mandatory/4-tax.js:23]: :+1:

**Review by jstadnik — COMMENTED** [review]
Good work on the mandatory exercises, seems like you're getting it!

File: mandatory/3-function-output.js (hunk 5 above)
Comment (jstadnik) [reply mandatory/3-function-output.js:14]: Concat also lets you do things like

```javascript
someWord.concat(" ", otherWord, " ", "and", "other stuff")
```

**Pull request closed without merging**